clap_mangen/tests/roff.rs

```diff
@@ -83,3 +83,10 @@ fn sub_subcommands_help() {
         common::assert_matches_path("tests/snapshots/sub_subcommand_help.roff", cmd.clone());
     }
 }
+
+#[test]
+fn derive_num_args() {
+    use clap::CommandFactory;
+    let cmd = common::DeriveNumArgs::command();
+    common::assert_matches_path("tests/snapshots/derive_num_args.bash.roff", cmd);
+}
```

Comment:
Our testing strategy depends on the answer to the following question from the issue

> So the question is if we should fix this in the derive or in the callers.

If the bug is in `clap_mangen`, then we should have a test for the specific builder behavior that exposes the problem.  If the bug is in the derive and we are doing a short-term workaround, we should have a generic derive test that covers a variety of cases to make sure that clap_mangen plays nicely with whatever the derive generates.

Reply:
My understanding of the discussion in #4443 is that the preferred fix is a refactor of `clap_mangen`, which suggests that the issue is here.

> then we should have a test for the specific builder behavior that exposes the problem.

I believe the added test does this.  Please let me know if I've misunderstood something.

Reply:
Oh, are you saying the derive behavior should be hand-implemented in terms of builder?  Sure, I can do that.

Reply:
Sorry to interject
> if the bug is in clap_mangen, then we should have a test for the specific builder behavior that exposes the problem. If the bug is in the derive and we are doing a short-term workaround, we should have a generic derive test that covers a variety of cases to make sure that clap_mangen plays nicely with whatever the derive generates.

When i was working on #4645, I noticed a lot of behavioral differences in the way text was formatted between Derive and Builder API's, that indicates inconsistency upstream from clap_mangen. (ex: Option names always being Capitalized vs Being Lower Case). All the ones I found, apart from the issues in #4443, seemed cosmetic and fixable by refactoring clap_mangen.